dts/arm/ti/cc13xx_cc26xx.dtsi

```diff
@@ -123,11 +123,12 @@
 			status = "disabled";
 		};
 
+		/* The RTC peripheral backs the kernel system clock and tick timer. */
 		rtc: rtc@40092000 {
-			compatible = "ti,cc13xx-cc26xx-rtc";
+			compatible = "ti,cc13xx-cc26xx-rtc-timer";
```

Comment:
@fgrandel Thank you for the PR, when counter or PPS support is added how would the node change, I am trying to correlate the description in the PR and the changes, here only the compatible seems to be changed?

Reply:
@vaishnavachath Not sure whether I understand your question... When something like PPS is upstreamed, then it will probably initially look like this:

```
rtc: rtc@40092000 {
  compatible = "ti,cc13xx-cc26xx-rtc";
  reg = <0x40092000 0x1000>;
  interrupts = <4 0>;  /* interrupt #20 = 4 + 16 */

  timer {
    compatible = "ti,cc13xx-cc26xx-rtc-timer";
    status = "okay"; /* the system clock timer is mandatory */
  };

  pps {
    compatible = "ti,cc13xx-cc26xx-rtc-pps";
    status = "disabled"
  };
};
```

This is a quite established pattern for flexible peripherals with multiple uses/drivers, see e.g. this one:

https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/st/f0/stm32f0.dtsi#L249

```
		timers3: timers@40000400 {
			compatible = "st,stm32-timers";
			reg = <0x40000400 0x400>;
			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000002>;
			resets = <&rctl STM32_RESET(APB1, 1U)>;
			interrupts = <16 0>;
			interrupt-names = "global";
			st,prescaler = <0>;
			status = "disabled";

			pwm {
				compatible = "st,stm32-pwm";
				status = "disabled";
				#pwm-cells = <3>;
			};

			counter {
				compatible = "st,stm32-counter";
				status = "disabled";
			};
		};
```

Also see this discussion re MFD devices that goes into quite some detail and links to even more detailed precursors: https://github.com/zephyrproject-rtos/zephyr/issues/48934

Reply:
@fgrandel , Thanks for clarifying, in #48934 the scheme mentioned is:
```
 bus {
       parent {
          child1 {
          }; 
          child2 {
          };
          child3 {
          };
       };
   };
```
Here you have updated the child1, but I did not see the parent node changes in the diff, in the above example also you mentioned (added some comments on where I am slightly confused):
```
rtc: rtc@40092000 {
  compatible = "ti,cc13xx-cc26xx-rtc"; /* parent device : I don't see this in your diff */
  reg = <0x40092000 0x1000>;
  interrupts = <4 0>;  /* interrupt #20 = 4 + 16 */

  timer {
    compatible = "ti,cc13xx-cc26xx-rtc-timer"; /* child device: in your diff when you changed the compatible this is the parent device now?? */
    status = "okay"; /* the system clock timer is mandatory */
  };

  pps {
    compatible = "ti,cc13xx-cc26xx-rtc-pps";
    status = "disabled"
  };
};
```

Reply:
(Cleared this up over Discord) The misunderstanding here was that my PR description and commit comment didn't make it clear enough that this was a preparative change that was meant to ensure that names would not have to be changed (and backwards compat broken) in the future but that the hierarchical devicetree would not be introduced until actually required.

Reply:
@fgrandel Thanks for the clarification in chat, looking at the commit message/PR description, I assumed that you are adding the parent node as well that was the confusion.